shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Tx.hs

```diff
@@ -109,51 +110,44 @@ import Shelley.Spec.Ledger.TxData
     WitVKey (..),
     witKeyHash,
   )
+import Shelley.Spec.Ledger.Value
 
 -- | Higher Kinded Data
 type family HKD f a where
   HKD Identity a = a
   HKD f a = f a
 
-data WitnessSetHKD f crypto = WitnessSet'
-  { addrWits' :: !(HKD f (Set (WitVKey crypto 'Witness))),
+data WitnessSetHKD f crypto v = WitnessSet'
+  { addrWits' :: !(HKD f (Set (WitVKey crypto v 'Witness))),
     msigWits' :: !(HKD f (Map (ScriptHash crypto) (MultiSig crypto))),
-    bootWits' :: !(HKD f (Set (BootstrapWitness crypto))),
+    bootWits' :: !(HKD f (Set (BootstrapWitness crypto v))),
     txWitsBytes :: LByteString
   }
 
-deriving instance Crypto crypto => Show (WitnessSetHKD Identity crypto)
+deriving instance Crypto crypto => Show (WitnessSetHKD Identity crypto v)
 
-deriving instance Crypto crypto => Eq (WitnessSetHKD Identity crypto)
+deriving instance Crypto crypto => Eq (WitnessSetHKD Identity crypto v)
 
-deriving instance Crypto crypto => Generic (WitnessSetHKD Identity crypto)
-
-deriving via
-  AllowThunksIn
-    '[ "txWitsBytes"
```

Comment:
don't we still want the annotations to be thunks?

Reply:
We couldnt get deriving via to work with `AllowThunksIn` because of lack of ability to derive `Generic` for everything in `Val`. Is this essential?

Reply:
If the field is lazy but the instance doesn't mark it as allowed, this will trip thunk detection.

If the field is made strict, this will trigger serialization every time a witness set is modified.

One solution would be to figure out a way to get a generic instance for the val. Another would be to write a manual instance.